### ISSUE TITLE
20.0.8 Maximum call stack size exceeded on run-many

### ISSUE THREAD

**Issue by yharaskrik** [nrwl/nx#28788]
### Current Behavior

I recently upgraded to 20.0.7 but then realized there was an issue with dependent projects, but then 20.0.8 was released the fix for that, just tried it this morning but now I am getting

```
$ nx run-many -t generate-schema --parallel

 NX   Maximum call stack size exceeded

Pass --verbose to see the stacktrace.

error: script "nx" exited with code 1

Exited with code exit status 1
```

I successfully ran another command prior to this in CI though so its not all commands (`nx run-many -t generate-prisma-client --parallel`).

Note: this ran find on 20.0.7 so the issue was introduced in 20.0.8

### Expected Behavior

Run-many works correctly.

### Steps to Reproduce

Unsure, would assume it has to do with the graph and possibly a circular dependency.

### Nx Report

```
 NX   Report complete - copy this into the issue template

Node           : 22.8.0
OS             : darwin-arm64
Native Target  : aarch64-macos
yarn           : 1.22.17

nx                      : 20.0.8
@nx/js                  : 20.0.8
@nrwl/js                : 19.8.3
@nx/jest                : 20.0.8
@nrwl/jest              : 19.8.3
@nx/linter              : 19.8.3
@nx/eslint              : 20.0.8
@nx/workspace           : 20.0.8
@nrwl/workspace         : 19.8.3
@nx/angular             : 20.0.8
@nrwl/angular           : 19.8.3
@nx/cypress             : 20.0.8
@nrwl/cypress           : 19.8.3
@nx/devkit              : 20.0.8
@nrwl/devkit            : 19.6.5
@nx/esbuild             : 20.0.8
@nrwl/esbuild           : 19.8.3
@nx/eslint-plugin       : 20.0.8
@nrwl/eslint-plugin-nx  : 19.8.3
@nx/express             : 20.0.8
@nrwl/express           : 19.8.3
@nx/nest                : 20.0.8
@nrwl/nest              : 19.8.3
@nx/node                : 20.0.8
@nrwl/node              : 19.8.3
@nx/plugin              : 20.0.8
@nrwl/nx-plugin         : 19.8.3
@nx/rspack              : 20.0.8
@nrwl/rspack            : 19.8.0
@nx/storybook           : 20.0.8
@nrwl/storybook         : 19.8.3
@nrwl/tao               : 19.6.5
@nx/vite                : 20.0.8
@nrwl/vite              : 19.8.3
@nx/web                 : 20.0.8
@nrwl/web               : 19.8.3
@nx/webpack             : 20.0.8
@nrwl/webpack           : 19.8.3
typescript              : 5.5.4
---------------------------------------
Registered Plugins:
./plugins/ci-test-targets.ts
./plugins/compile-css.plugin.ts
./plugins/dotenv-vault.plugin.ts
./plugins/fly.plugin.ts
---------------------------------------
Community plugins:
@ngneat/spectator        : 18.0.2
@ngrx/component          : 18.1.1
@ngrx/component-store    : 18.1.1
@ngrx/data               : 18.1.1
@ngrx/effects            : 18.1.1
@ngrx/entity             : 18.1.1
@ngrx/operators          : 18.1.1
@ngrx/router-store       : 18.1.1
@ngrx/store              : 18.1.1
@ngrx/store-devtools     : 18.1.1
@skyux/i18n              : 10.23.0
@skyux/packages          : 10.23.0
@storybook/angular       : 8.3.3
@testing-library/angular : 17.3.1
angular-calendar         : 0.31.0
apollo-angular           : 7.1.2
ngxtension               : 4.0.0
---------------------------------------
Local workspace plugins:
         @trellis/plugins/apollo-graphql
         @trellis/plugins/nest-graphql
         @trellis/plugins/aws-deploy
         @trellis/plugins/repository
         @trellis/plugins/angular
         @trellis/plugins/docker
         @trellis/plugins/dotenv
         @trellis/plugins/prisma
         @trellis/plugins/jest
         @trellis/plugins/node
         @trellis/plugins/fly
---------------------------------------
The following packages should match the installed version of nx
  - @nrwl/js@19.8.3
  - @nrwl/jest@19.8.3
  - @nx/linter@19.8.3
  - @nrwl/workspace@19.8.3
  - @nrwl/angular@19.8.3
  - @nrwl/cypress@19.8.3
  - @nrwl/devkit@19.6.5
  - @nrwl/esbuild@19.8.3
  - @nrwl/eslint-plugin-nx@19.8.3
  - @nrwl/express@19.8.3
  - @nrwl/nest@19.8.3
  - @nrwl/node@19.8.3
  - @nrwl/nx-plugin@19.8.3
  - @nrwl/rspack@19.8.0
  - @nrwl/storybook@19.8.3
  - @nrwl/tao@19.6.5
  - @nrwl/vite@19.8.3
  - @nrwl/web@19.8.3
  - @nrwl/webpack@19.8.3

To fix this, run `nx migrate nx@20.0.8`
```

### Failure Logs

```
$ nx run-many -t generate-schema --parallel

 NX   Maximum call stack size exceeded

Pass --verbose to see the stacktrace.

error: script "nx" exited with code 1

Exited with code exit status 1
```

### Package Manager Version

bun@latest

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Was working on 20.0.7, broke on 20.0.8

**Comment by jaysoo** [comment]
Likely caused by https://github.com/nrwl/nx/pull/28669

**Comment by jaysoo** [comment]
@yharaskrik Also note that you can safely remove the `@nrwl/` scoped packages as those have been replaced by `@nx/` packages.

**Comment by yharaskrik** [comment]
> [@yharaskrik](https://github.com/yharaskrik) Also note that you can safely remove the `@nrwl/` scoped packages as those have been replaced by `@nx/` packages.

Sounds good thanks, I don't actually have them in my package.json anymore, must still just exist in node_modules.

**Comment by xiongemi** [comment]
@yharaskrik it is hard to debug without a repo. Could you generate a project graph or task graph for generate-schema using `nx graph`? does it also throw the error "Maximum call stack size exceeded" when you try to view the task graph for generate-schema?

**Comment by yharaskrik** [comment]
> [@yharaskrik](https://github.com/yharaskrik) it is hard to debug without a repo. Could you generate a project graph or task graph for generate-schema using `nx graph`? does it also throw the error "Maximum call stack size exceeded" when you try to view the task graph for generate-schema?

Ya, sorry about that, I know repros are preferred.

I will check this out for you right now!

**Comment by yharaskrik** [comment]
> [@yharaskrik](https://github.com/yharaskrik) it is hard to debug without a repo. Could you generate a project graph or task graph for generate-schema using `nx graph`? does it also throw the error "Maximum call stack size exceeded" when you try to view the task graph for generate-schema?

`nx graph` successfully opens the UI but when I go to tasks and filter by I get this error.

![Image](https://github.com/user-attachments/assets/6140891d-514c-4e2c-a772-42e3a2f4df83)

I looked in the daemon.log but I didn't see anything in there, is there somewhere I could potentially pull a stack trace for you?

**Comment by xiongemi** [comment]
i published a pr release 0.0.0-pr-28793-3c45083. could you help me to verify whether this issue is fixed in my pr?

**Comment by yharaskrik** [comment]
> i published a pr release 0.0.0-pr-28793-3c45083. could you help me to verify whether this issue is fixed in my pr?

All fixed! 🎉

**Comment by DaSchTour** [comment]
How can I still the pr release? I have the same issue an also would like to check if the changes help.
